tests/test_anf_ir_utils.py

```diff
@@ -18,3 +18,26 @@ def test_dfs_graphs():
     value = Apply([in0], Graph())
     assert set(dfs(value)) == {value, in0}
     assert set(dfs(value, follow_graph=True)) == {value, in0, in1}
+
+
+def test_toposort():
+    g0 = Graph()
+    g0.output = Constant(1)
+    g1 = Graph()
+    in0 = Constant(g0)
+    value = Apply([in0], g1)
+    g1.output = value
+    assert list(toposort(g1)) == [in0, value, g1.return_.inputs[0], g1.return_]
```

Comment:
Suggestion: in addition to checking the list explicitly, you can automatically check that the sort is correct with this function:

```python
def _check_toposort(order, root):
    nodes = set(dfs(root))
    assert len(order) == len(nodes)
    for node in nodes:
        for i in node.inputs:
            assert order.index(i) < order.index(node)
```